### PR TITLE
Add logic to convert youtube urls into embed

### PR DESCRIPTION
**Pull request by Mehdi-Miah**

### CODE REVIEW

**Comment by Mehdi-Miah** [comment]
one question, if this change is applied to guerrilla via an upgrade. Will the filter be immediately available? the startup.cs is only run once? or will it run when the new widgets module is added?

File: Filters/EmbedUrlFilter.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+{
+    public class EmbedUrlFilter : ILiquidFilter
+    {
+        public ValueTask<FluidValue> ProcessAsync(FluidValue input, FilterArguments arguments, LiquidTemplateContext context)
+        {
+            var videoUrl = input.ToObjectValue().ToString();
+
+            if (videoUrl.Contains("youtube.com/watch?v="))
+            {
+                videoUrl = ConvertToEmbed(videoUrl);
```
Comment (peterkeating) [inline Filters/EmbedUrlFilter.cs:16]: This assignment isn't needed.

```
return new ValueTask<FluidValue>(new StringValue(ConvertToEmbed(videoUrl)));
```

**Review by peterkeating — COMMENTED** [review]
Can you address the code smells picked up by SonarCloud too.

File: Filters/EmbedUrlFilter.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            }
+
+            return new ValueTask<FluidValue>(new StringValue(videoUrl));
+        }
+
+        public static string ConvertToEmbed(string url)
+        {
+            url = url.Replace("watch?v=", "embed/");
+
+            if (url.IndexOf("&") >= 0)
```
Comment (Mehdi-Miah) [inline Filters/EmbedUrlFilter.cs:26]: i removed the check .Contains("&"), IndexOf() can both check and return the index if it exists right?

**Review by peterkeating — APPROVED** [review]

**Pull request merged**